test/unit/spec/narrative_core/narrativeViewersSpec.js

```diff
@@ -0,0 +1,139 @@
+/* eslint-env jasmine */
+define ([
+    'narrativeViewers',
+    'narrativeConfig'
+], (
+    Viewers,
+    Config
+) => {
+    'use strict';
+
+    // data format taken from
+    // https://github.com/kbase/narrative_method_store/blob/master/NarrativeMethodStore.spec#L709
+    // and modified to what's needed for running the module and testing
+    const categoryData = [
+        {},
+        {
+            'Viewers/a_viewer_app': {},
+            'Viewers/error_viewer_app': {
+                loading_error: 'Not a real app'
+            }
+        },
+        {},
+        {
+            'Module.Type1': {
+                view_method_ids: ['Viewers/a_viewer_app'],
+                landing_page_url_prefix: 'type1',
+                name: 'Type 1'
+            },
+            'Module.Type2': {
+                view_method_ids: ['Viewers/missing_viewer_app'],
+                landing_page_url_prefix: 'type2',
+                name: 'Type 2'
+            },
+            'Module.Type3': {
+                view_method_ids: ['Viewers/error_viewer_app'],
+                landing_page_url_prefix: 'type3',
+                name: 'Type 3'
+            }
```

Comment:
is this data used/tested anywhere?

Reply:
Yes, that's the mocked result from NMS.list_categories (the mocked out type information)